includes/import/abstract-wc-product-importer.php

```diff
@@ -0,0 +1,941 @@
+<?php
+/**
+ * Abstract Product importer
+ *
+ * @author   Automattic
+ * @category Admin
+ * @package  WooCommerce/Import
+ * @version  3.1.0
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Include dependencies.
+ */
+if ( ! class_exists( 'WC_Importer_Interface', false ) ) {
+	include_once( WC_ABSPATH . 'includes/interfaces/class-wc-importer-interface.php' );
+}
+
+/**
+ * WC_Product_Importer Class.
+ */
+abstract class WC_Product_Importer implements WC_Importer_Interface {
+
+	/**
+	 * CSV file.
+	 *
+	 * @var string
+	 */
+	protected $file = '';
+
+	/**
+	 * Importer parameters.
+	 *
+	 * @var array
+	 */
+	protected $params = array();
+
+	/**
+	 * Raw keys - CSV raw headers.
+	 *
+	 * @var array
+	 */
+	protected $raw_keys = array();
+
+	/**
+	 * Mapped keys - CSV headers.
+	 *
+	 * @var array
+	 */
+	protected $mapped_keys = array();
+
+	/**
+	 * Raw data.
+	 *
+	 * @var array
+	 */
+	protected $raw_data = array();
+
+	/**
+	 * Parsed data.
+	 *
+	 * @var array
+	 */
+	protected $parsed_data = array();
+
+	/**
+	 * Get file raw headers.
+	 *
+	 * @return array
+	 */
+	public function get_raw_keys() {
+		return $this->raw_keys;
+	}
+
+	/**
+	 * Get file mapped headers.
+	 *
+	 * @return array
+	 */
+	public function get_mapped_keys() {
+		return $this->mapped_keys;
+	}
+
+	/**
+	 * Get raw data.
+	 *
+	 * @return array
+	 */
+	public function get_raw_data() {
+		return $this->raw_data;
+	}
+
+	/**
+	 * Get parsed data.
+	 *
+	 * @return array
+	 */
+	public function get_parsed_data() {
+		return apply_filters( 'woocommerce_product_parsed_data', $this->parsed_data, $this->get_raw_data() );
+	}
+
+	/**
+	 * Process a single item and save.
+	 *
+	 * @param  array $data Raw CSV data.
+	 * @return WC_Product|WC_Error
+	 */
+	protected function process_item( $data ) {
+		// Ignore IDs and create new products.
+		// @todo Mike said that we should have something to force create.
```

Comment:
Just to be clear I think we should have a checkbox to enable 'merge'.

If merge is on, IDs are mapped and non-matching IDs are ignored.

If merge is off, products are imported not merged. IDs are ignored.

@claudiulodro ?